REL-899797 update readme (#12)
relativitydev/relativity-transfer-sdk-samples: README.md

The library supports `.NETStandard 2.0`, which means it is cross-platform. You can run it on Windows or Linux!

## Samples
#### Repository structure:
The repository contains 3 projects:
- `Relativity.Transfer.SDK.Samples.Core` - contains all the interfaces, models, UI, and helpers used by the `Relativity.Transfer.SDK.Samples.Repository` project.
- `Relativity.Transfer.SDK.Samples.Repository` - contains all the samples.
- `Relativity.Transfer.SDK.Samples.CLI` - contains a command-line interface for the `Relativity.Transfer.SDK.Samples.Repository` project. It uses IoC container to create instances of required classes.

#### Examples structure:

> - All code presenting a particular example is contained only in a single file (if additional dependencies are required they are injected by IoC).

Reviewer: Using of IoC contradicts the statement that "everything is one file". Maybe:
```suggestion
- Code presenting a particular example is contained in a single file (see the table below to match a sample with a code file). If additional dependencies are required they are injected by IoC [here the rough location, when this IoC is defined].
```

        - Note: This value can be taken from Relativity. Find the `Servers` tab, filter it by `Fileshare`, search for the specified one, and copy its `UNC` path value **without** the `\Files\` suffix.
        - **Note**: It is crucial to remember that backslash `\` characters should be escaped using a double backslash `\\` (this is required by json file to properly deserialize a path).
    - `Common.FileShareRelativePath` - The location where the files are transferred relative to the root of the fileshare.
        - **Note**: `<transfer job id>` folder is created in provided location.
        - **Note**: Provided path can not have `\` at the beginning.
        - The path must be rooted in one of the core folders that reside on the fileshare (like Files, Temp, etc.)
        - **Note**: It is crucial to remember that backslash `\` characters should be escaped using a double backslash `\\` (this is required by json file to properly deserialize a path).
    - `Common.OAuthCredentials` - The OAuth2 secret ID and secret used to authenticate in RelativityOne (Find the `OAuth2 Client` tab, then find the user and copy `ClientId` and `ClientSecret` fileds).

> - The sttings specified for a sample are commented in `appsettings.json` file. Which section is responsible for which sample is described in the table above.

Reviewer: ```suggestion
- The settings specified for a sample are commented in `appsettings.json` file. Which section is responsible for which sample is described in the table above.
```

## Authentication
- First, the transfer client object has to be created, which is used to manage transfers:
```cs
// FullPathWorkflow
var transferClient = TransferClientBuilder.FullPathWorkflow
    .WithAuthentication(authenticationProvider)
    .WithClientName(clientName)
    .Build();

// or JobBasedWorkflow
var transferClient = TransferClientBuilder.FullPathWorkflow
    .WithAuthentication(authenticationProvider)
    .WithClientName(clientName)
    .Build();
```

> - `TransferSKD` uses a bearer token in order to authenticate the transfer.

Reviewer: ```suggestion
- `TransferSDK` uses a bearer token in order to authenticate the transfer.
```

> - To pass the token, the `Relativity.Transfer.SDK.Interfaces.Authentication.IRelativityAuthenticationProvider` must be implemented and an instance should be passed to the clint. Sample implementation of authentication provider is [here](https://github.com/relativitydev/relativity-transfer-sdk-samples/blob/main/Source/Relativity.Transfer.SDK.Samples.Core/Authentication/RelativityAuthenticationProvider.cs). 

Reviewer: ```suggestion
- To pass the token, the `Relativity.Transfer.SDK.Interfaces.Authentication.IRelativityAuthenticationProvider` must be implemented and an instance should be passed to the client. Sample implementation of authentication provider is [here](https://github.com/relativitydev/relativity-transfer-sdk-samples/blob/main/Source/Relativity.Transfer.SDK.Samples.Core/Authentication/RelativityAuthenticationProvider.cs). 
```

> - In order to get the token, [Bearer token authentication](https://platform.relativity.com/RelativityOne/Content/REST_API/REST_API_authentication.htm#_Bearer_token_authentication) is used, which requires user OAuth2 client id and client secret. (see [SampleBearerTokenRetriver](https://github.com/relativitydev/relativity-transfer-sdk-samples/blob/main/Source/Relativity.Transfer.SDK.Samples.Core/Authentication/BearerTokenRetriever.cs))

Reviewer: ```suggestion
- In order to get the token, [Bearer token authentication](https://platform.relativity.com/RelativityOne/Content/REST_API/REST_API_authentication.htm#_Bearer_token_authentication) is used, which requires user's OAuth2 client id and client's secret. (see [SampleBearerTokenRetriver](https://github.com/relativitydev/relativity-transfer-sdk-samples/blob/main/Source/Relativity.Transfer.SDK.Samples.Core/Authentication/BearerTokenRetriever.cs))
```

        - **Note**: `<transfer job id>` folder is created in provided location.
        - **Note**: Provided path can not have `\` at the beginning.
        - The path must be rooted in one of the core folders that reside on the fileshare (like Files, Temp, etc.)
        - **Note**: It is crucial to remember that backslash `\` characters should be escaped using a double backslash `\\` (this is required by json file to properly deserialize a path).
    - `Common.OAuthCredentials` - The OAuth2 secret ID and secret used to authenticate in RelativityOne (Find the `OAuth2 Client` tab, then find the user and copy `ClientId` and `ClientSecret` fileds).
- The settings specified for a sample are commented in `appsettings.json` file. Which section is responsible for which sample is described in the table above.

## Authentication
- First, the transfer client object has to be created, which is used to manage transfers:
```cs
// FullPathWorkflow
var transferClient = TransferClientBuilder.FullPathWorkflow
    .WithAuthentication(authenticationProvider)
    .WithClientName(clientName)
    .Build();

// or JobBasedWorkflow

> var transferClient = TransferClientBuilder.FullPathWorkflow

Reviewer 2: Change to JobBasedWorkflow

Author: done
